org.ultimatepp.TheIDE.yml

```diff
@@ -0,0 +1,113 @@
+      - 3p/umk/umk app/,3p/uppsrc UppTerm 3p/umk/GCC.bm -brv +GUI,SHARED upp-term
+      - install -Dm755 upp-term /app/bin/upp-term
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/UppTerm/archive/refs/tags/v2023.1.tar.gz
+        sha256: a40da7148d16f8f7c9989da1ad430d69c09483c91dab0607bc422ae6c184dda2
+      - type: archive
+        url: https://github.com/ismail-yilmaz/Terminal/archive/refs/tags/2023.2.tar.gz
+        sha256: 320d8574842825d1d8ddfa46addcb4eb03a9120116caea192582f2a02d26ec09
+        dest: 3p/Terminal
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/uppsrc-17045.tar.gz
+        sha256: 23793a363a140139728a30cc050c35287d6ff97ee03125d62f8ad8acf06e40af
+        dest: 3p/uppsrc
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/umk-17045-linux-x86-64.tar.gz
+        sha256: 36ee1a061590f73adf6367ef85c51adbfff164dd53a53480ae82eb0b9925b923
+        only-arches: [x86_64]
+        dest: 3p/umk
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/umk-17045-linux-aarch64.tar.gz
+        sha256: 2943ee5e044f3ff41393e921188b8b5a89cfca3ecaf1bfef66afb3a65882766c
+        only-arches: [aarch64]
+        dest: 3p/umk
+
+  - name: run-ide
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 run-ide.sh /app/bin/run-ide.sh
+      - install -D TheIDE_512x512.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_256x256.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_128x128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_64x64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -D org.ultimatepp.TheIDE.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -D org.ultimatepp.TheIDE.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: file
+        path: run-ide.sh
+      - type: file
+        path: org.ultimatepp.TheIDE.desktop
+      - type: file
+        path: org.ultimatepp.TheIDE.metainfo.xml
+      - type: file
+        path: res/TheIDE_512x512.png
+      - type: file
+        path: res/TheIDE_256x256.png
+      - type: file
+        path: res/TheIDE_128x128.png
+      - type: file
+        path: res/TheIDE_64x64.png
+
+  - name: host-spawn
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 host-spawn /app/bin/host-spawn
+    sources:
+      - type: file
+        url: https://github.com/1player/host-spawn/releases/download/1.5.0/host-spawn-x86_64
+        sha256: dbf67e7e111c4fe1edb0c642cbb4193064ca5b384aeb1054fc2befba6ed88b83
+        only-arches: [x86_64]
+        dest-filename: host-spawn
+      - type: file
+        url: https://github.com/1player/host-spawn/releases/download/1.5.0/host-spawn-aarch64
```

Comment:
can't these be built from source?

Reply:
Hello @hfiguiere,

The above statement downloads a static executable written in Go that works on a given operating system and architecture. So, there are no real benefits to building from scratch. Also, the author of the package provides pre-built binaries for Linux, x86_64, and arrch64 architectures. Which are the only architectures supported by Flathub at the momenet.

Currently, on the Flathub, there are two approaches to using host-spawn:
- https://github.com/flathub/com.visualstudio.code/blob/c2704e2c9c31ba12f8266fc27d5c73f6e68a51f4/com.visualstudio.code.yaml#L138 (VS Code - Downloading static executable)
- https://github.com/flathub/org.kde.qmlkonsole/blob/fe725223c81bd050da8b4dc96c10e17e8214a454/org.kde.qmlkonsole.json#L21 (QML Konsole - Downloading source package and building from scratch)

Personally, I think the first option is much more straightforward and relatively easy to maintain. You do not need to manually add all GoLang dependencies on your own each time you want to upgrade a package.

Please let me know what your recommendation is here. I can rework to build from sources if it is needed. However, I would like to avoid this in this particular case when everything that is needed is provided directly in a binary package. Also, looking at the current application distributed via Flathub, there is no unified way to do it.

Reply:
Hello @hfiguiere,

I manage to build host-spawn from sources. It was not straightforward, but in the end it was possible. I made decision to patch host-spawn, because it was using golang.org/x/sys in the unknown version (commit). So, I decided to change it to v0.15.0. The patch is under review in the upstream (https://github.com/1player/host-spawn/pull/30). Please keep in mind that using version from commit might be treated as security vulnerability, so it must be avoided whenever possible. I hope author will soon accept my PR and release new version. As far as I remember it was also the Flathub recommendation to build only from tagged releases.

Klugier